### ISSUE TITLE
Add normalized fields

### ISSUE THREAD

**Issue by vbohata** [elastic/ecs#264]
There are several fields expecting to contain normalized fields like:
- event.kind
- event.category
 ...

When I try to add some log source I do as much normalization/field naming as I can. But usually I get undocumented message format or hundereds of wrongly documented fields from our source systems. It is about trial and error to do some mapping. Usually it is impossible to know all the available values these fields can contain. There are definitely cases where I need to map these fields to event.kind, event.category etc. as these fields are appropriate but can not guarantee what values they can contain. So  event.kind can be "event" or "event-warn" or ... and it is fine I need to map these original values to event.kind.

So I propose to create some universal way to allow arbitrary field values together with normalized alternatives if possible.

Something like:
- event.kind: event-sys1
- event.norm.kind: event
- event.category: userman/del
- event.norm.category: user-management

So norm fields should contain normalized fields. From my experience it is absolutely impossible to do some normalization/set of allowed values in general. Each system has its own values, each software update can change these values and as you have tens of these systems, no one can keep track of these changes.

**Comment by webmat** [comment]
Yes, we're aware that doing event categorization is going to be a big challenge, and even that it's not always possible to know in advance all the values we'll get from our sources.

We're currently working on this, and will come out with more specific recommendations soon.

As usual, there's a few approaches we can take. The "default" field could contain the raw data, and the normalized data goes into another field (as you suggest). Or we could instead make the "default" field contain the normalized data, and the raw information goes into another field instead (e.g. `event.raw.kind`).

Our current recommendation about the default fields doesn't mean we will take this approach in the end. It gives us the possibility of opening up their use for the raw value in the future, and introduce a place for normalized values. Essentially we'd be removing a constraint if we go this direction.

We'll take your feedback into account when working on this further :-)

In the meantime, I would say it's safer to define fields for your raw values and not use the fields where we say we'll introduce guidelines. If you're comfortable using these fields, then adjusting them when we come out with specific guidelines, you're free to do so as well.
